src/test/java/com/exasol/adapter/dialects/ScalarFunctionsParameterCache.java

```diff
@@ -0,0 +1,62 @@
+package com.exasol.adapter.dialects;
+
+import java.io.*;
+import java.util.*;
+
+import org.yaml.snakeyaml.Yaml;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This class caches the parameters for runs that were passed the tests. By that in upcoming test runs can only test
+ * these runs and by that are a lot faster.
+ */
+public class ScalarFunctionsParameterCache {
+    private static final String CACHE_FILE_NAME = "src/test/resources/scalarFunctionsParameterCache.yml";
+    private final Map<String, List<String>> parameterCache;
+    private final Yaml yaml;
+
+    public ScalarFunctionsParameterCache() {
+        try {
+            this.yaml = new Yaml();
+            this.parameterCache = loadParameterCache();
+        } catch (final FileNotFoundException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-PGVS-10")
+                    .message("Failed to read scalar functions from parameters cache.").toString(), exception);
+        }
+    }
+
+    private Map<String, List<String>> loadParameterCache() throws FileNotFoundException {
+        final Map<String, List<String>> loadedData = this.yaml.load(new FileReader(CACHE_FILE_NAME));
+        if (loadedData != null) {
+            return new TreeMap<>(loadedData);
+        } else {
+            return new TreeMap<>();
```

Comment:
Collections.emptyMap() ?

Reply:
I need a modifyable Map here, since I will add items. Collections.empty map() is unmodifiable. Apart from that I need a TreeMap so that the output file is sorted.